src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,252 @@
+     *
+     * @return \Memcached
+     */
+    public function setupConnection(array $servers = array(), array $options = array())
+    {
+        foreach ($servers as $server) {
+            $this->addServer($server);
+        }
+
+        foreach ($options as $opt => $val) {
+            $this->setOption($opt, $val);
+        }
+
+        return $this->memcached;
+    }
+
+    /**
+     * @param mixed[] $server
+     */
+    public function addServer(array $server)
+    {
+        list($host, $port, $weight) = $this->filterServerValues(
+            isset($server[0]) ? $server[0] : self::DEFAULT_HOST,
+            isset($server[1]) ? $server[1] : self::DEFAULT_PORT,
+            isset($server[2]) ? $server[2] : self::DEFAULT_WEIGHT
+        );
+
+        if (!$this->isServerAlreadyEnabled($host, $port)) {
+            $this->memcached->addServer($host, $port, $weight);
+        }
+    }
+
+    /**
+     * @param mixed $opt
+     * @param mixed $val
+     *
+     * @return bool
+     */
+    public function setOption($opt, $val)
+    {
+        list($opt, $val) = $this->filterOptionValues($opt, $val);
+
+        return $this->memcached->setOption($opt, $val)
+            && $this->memcached->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    /**
+     * @param mixed $opt
+     * @param mixed $val
+     *
+     * @return mixed[]
+     */
+    private function filterOptionValues($opt, $val)
+    {
+        list($opt, $val) = array(
+            $opt = $this->tryResolvingConstant($opt),
+            $val = $this->tryResolvingConstant($val),
+        );
+
+        if (false === filter_var($opt, FILTER_VALIDATE_INT)) {
+            throw new InvalidArgumentException(sprintf('Option type invalid "%s": must resolve as int', $opt));
+        }
+
+        if (false === filter_var($val, FILTER_VALIDATE_INT) && false === filter_var($val, FILTER_VALIDATE_BOOLEAN)) {
+            throw new InvalidArgumentException(sprintf('Option value invalid "%s": must resolve as int or bool', $val));
+        }
+
+        return array($opt, $val);
+    }
+
+    /**
+     * @param mixed $val
+     *
+     * @return mixed
+     */
+    private function tryResolvingConstant($val)
+    {
+        return defined($constant = '\Memcached::'.$val) ? constant($constant) : $val;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        foreach ($this->memcached->getMulti($ids) as $id => $val) {
+            try {
+                yield $id => parent::unserialize($val);
+            } catch (\Exception $e) {
+                continue;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id)
+    {
+        return $this->memcached->get($id) !== false
+            && $this->memcached->getResultCode() !== \Memcached::RES_NOTFOUND;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doClear($namespace)
+    {
+        return $this->memcached->flush()
```

Comment:
If possible, it would be better to delete only the keys that start with `$namespace`.
Looking at https://gist.github.com/stetic/2929166 (Ruby) or http://stackoverflow.com/a/34125587 (memcache~~d~~ ext.) there seem to be a way to do so properly.
Could you please try how we could to something similar with the memcached ext.?